modules/csg/editor/csg_gizmos.cpp

```diff
@@ -393,9 +516,36 @@ void CSGShape3DGizmoPlugin::redraw(EditorNode3DGizmo *p_gizmo) {
 	}
 }
 
+void EditorPluginCSG::edit(Object *p_object) {
+	CSGShape3D *csg_shape = Object::cast_to<CSGShape3D>(p_object);
+	if (csg_shape && csg_shape->is_root_shape()) {
+		csg_shape_editor->edit(csg_shape);
+	} else {
+		csg_shape_editor->edit(nullptr);
+	}
+}
+
+bool EditorPluginCSG::handles(Object *p_object) const {
+	CSGShape3D *csg_shape = Object::cast_to<CSGShape3D>(p_object);
+	return csg_shape && csg_shape->is_root_shape();
+}
+
+void EditorPluginCSG::make_visible(bool p_visible) {
+	if (p_visible) {
+		csg_shape_editor->options->show();
+	} else {
+		csg_shape_editor->options->hide();
+		csg_shape_editor->edit(nullptr);
```

Comment:
This should be unnecessary, because editor already calls `edit(nullptr)` when the plugin can stop editing.

Reply:
removed